packages/babel-parser/src/parser/statement.js

```diff
@@ -1563,6 +1567,24 @@ export default class StatementParser extends ExpressionParser {
       }
 
       this.checkGetterSetterParams(publicMethod);
+    } else if (
+      isContextual &&
+      key.name === "accessor" &&
+      !this.isLineTerminator()
```

Comment:
Probably this branch may also be merged with the following one.
```suggestion
      isContextual &&
      key.name === "accessor" &&
      this.hasPlugin("decorators") &&
      !this.isLineTerminator()
```

Reply:
Same as https://github.com/babel/babel/pull/13681#discussion_r689425353

Reply:
I tried to merge the two branches, but it results in some nested `if` logic that actually gets fairly confusing. If you want me to do it I can still, but I think it's much clearer this way (if a little more verbose) so decided to keep it.

Reply:
Just keep it as is then.